outbox.js

```diff
@@ -1,310 +1,242 @@
     } else {
-      postUrl = this.activityData.object;
-    }
-
-    try {
-      const response = await fetch(postUrl);
-      if (!response.ok) {
-        throw new Error(`HTTP error! Status: ${response.status}`);
-      }
-      const postData = await response.json();
-      // Determine how to extract content based on the post source
-      const content = isDirectPost ? postData.content : postData.object.content;
-      this.displayPostContent(content, postUrl);
-    } catch (error) {
-      console.error("Error fetching post content:", error);
+      postUrl = this.activityData.object
     }
+    this.displayPostContent(postUrl)
   }
 
-  displayPostContent(content, url) {
+  displayPostContent (url) {
     // Clear existing content
-    this.innerHTML = "";
+    this.innerHTML = ''
 
     // Create and append the distributed-post component
-    const distributedPostElement = document.createElement("distributed-post");
-    distributedPostElement.setAttribute("url", url);
-    this.appendChild(distributedPostElement);
+    const distributedPostElement = document.createElement('distributed-post')
+    distributedPostElement.setAttribute('url', url)
+    this.appendChild(distributedPostElement)
   }
 
-  fetchAndUpdatePost(activityData) {
-    let postUrl;
-    // Determine the source of the post (direct activity or URL pointing to the activity)
-    const isDirectUpdate =
-      typeof activityData.object === "string" ||
-      activityData.object instanceof String;
-
-    if (isDirectUpdate) {
-      // If it's a direct update, use the URL from the 'object' property
-      postUrl = activityData.object;
-    } else if (activityData.object && activityData.object.id) {
-      // If the 'object' property contains an 'id', use it as the URL
-      postUrl = activityData.object.id;
-    } else {
-      // Otherwise, use the 'id' property of the activityData itself
-      postUrl = activityData.id;
-    }
-
-    this.fetchAndDisplayPost(postUrl);
+  displayUnimplemented () {
+    const message = `Activity type ${this.activityType} is not implemented yet.`
+    const messageElement = document.createElement('p')
+    messageElement.textContent = message
+    this.appendChild(messageElement)
   }
 
-  renderActivity() {
+  renderActivity () {
     // Clear existing content
-    this.innerHTML = "";
+    this.innerHTML = ''
 
     switch (this.activityType) {
-      case "Create":
-        this.fetchAndDisplayPost();
-        break;
-      case "Update":
-        this.fetchAndUpdatePost(this.activityData);
-        break;
-      case "Follow":
-        this.displayFollowActivity();
-        break;
-      case "Like":
-        this.displayLikeActivity();
-        break;
+      case 'Create':
+        this.fetchAndDisplayPost()
+        break
+      case 'Update':
+        this.fetchAndDisplayPost()
+        break
+      case 'Announce':
+        // TODO: Add UI saying this was "reposted"
+        this.fetchAndDisplayPost()
```

Comment:
We should add a new function for this that has the info of the actor that "reposted" the note.